google/cloud/opentelemetry/internal/recordable.cc

```diff
@@ -160,6 +160,52 @@ google::devtools::cloudtrace::v2::Span::SpanKind MapSpanKind(
   }
 }
 
+template <typename T>
+opentelemetry::common::AttributeValue convertVector(std::vector<T> const& v) {
+  return opentelemetry::nostd::span<T const>(v.data(), v.size());
+}
+template <>
+opentelemetry::common::AttributeValue convertVector(std::vector<bool> const&) {
+  // We discard the data in vector<bool> attributes because:
+  // - We cannot convert a vector<bool> to a span<bool> without allocating
+  //   memory.
+  // - The exporter drops span<bool> attributes anyways.
```

Comment:
I don't buy the first reason, particularly in view of the third reason.

I buy the second reason.  But it does create a trap if we ever fix the exporter 🤷